src/smart_review/ai/base.py

```diff
@@ -0,0 +1,208 @@
+import logging
+import time
+from abc import ABC, abstractmethod
+from datetime import datetime
+from enum import Enum
+from typing import Any, Dict, List, Tuple
+
+from attrs import define, field
+from github.PullRequestReview import PullRequestReview
+
+from smart_review.exceptions import SmartReviewGithubException
+from smart_review.gitops.github import GitHubClient
```

Comment:
## Smart Review

The method 'process_request' lacks proper error handling which could lead to runtime exceptions if the input is not as expected.